### PR TITLE
Porting automated tasks and test cases into thin-edge.io repository

### PR DESCRIPTION
**Pull request by gligorisaev**
## Proposed changes

Robot Framework automated task and test cases have been stored in an private repository as POC, with this PR they will be ported to the main branch of the project.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further comments
Automated task added: 

- `build_install_thinedge.robot`

Automated test cases added:

- child_alarms
	- `c8y_child_alarms_rpi.robot`
- config_management
	- `inotify_crate.robot`
- customizing
	- `log_path_config.robot`
- MQTT_health_check
	- `health_c8y-configuration-plugin.robot`
	- `health_c8y-log-plugin.robot`
	- `health_tedge_mapper_c8y.robot`
	- `health_tedge-agent.robot`
	- `health_tedge-mapper-az.robot`
	- `health_tedge-mapper-collectd.robot`
- plugin_apt
	- `improve_tedge_apt_plugin_error_messages.robot`

### CODE REVIEW

File: tests/RobotFramework/plugin_apt/improve_tedge_apt_plugin_error_messages.robot [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+Library    DateTime
+Library    CryptoLibrary    variable_decryption=True
+Suite Setup            Open Connection And Log In
+Suite Teardown         SSHLibrary.Close All Connections
+
+*** Variables ***
+${HOST}           
+${USERNAME}       pi
+${PASSWORD}       crypt:LO3wCxZPltyviM8gEyBkRylToqtWm+hvq9mMVEPxtn0BXB65v/5wxUu7EqicpOgGhgNZVgFjY0o=  
+${VERSION}        0.7.4
```
Comment (rina23q) [inline tests/RobotFramework/plugin_apt/improve_tedge_apt_plugin_error_messages.robot:16]: I feel it is problematic in the future. When we release new version, we always need to update this hard-coded version?

File: tests/RobotFramework/plugin_apt/improve_tedge_apt_plugin_error_messages.robot [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    [Documentation]    Setting file name according architecture
+    ${FILENAME}    Set Variable    debian-packages-aarch64-unknown-linux-gnu
+    Log    ${FILENAME}
+    Set Global Variable    ${FILENAME}
+armv7
+    [Documentation]    Setting file name according architecture
+    ${FILENAME}    Set Variable    debian-packages-armv7-unknown-linux-gnueabihf
+    Log    ${FILENAME}
+    Set Global Variable    ${FILENAME}
+
```
Comment (rina23q) [inline tests/RobotFramework/plugin_apt/improve_tedge_apt_plugin_error_messages.robot:65]: No new line at the end of line

File: tests/RobotFramework/tasks/build_install_thinedge.robot [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    [Documentation]    Setting file name according architecture
+    ${FILENAME}    Set Variable    debian-packages-aarch64-unknown-linux-gnu
+    Log    ${FILENAME}
+    Set Global Variable    ${FILENAME}
+armv7
+    [Documentation]    Setting file name according architecture
+    ${FILENAME}    Set Variable    debian-packages-armv7-unknown-linux-gnueabihf
+    Log    ${FILENAME}
+    Set Global Variable    ${FILENAME}
+
```
Comment (rina23q) [inline tests/RobotFramework/tasks/build_install_thinedge.robot:162]: New line

**Review by rina23q — COMMENTED** [review]
Finish reviewing except README. I will review the README tomorrow.

File: tests/RobotFramework/README.md [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+		1. Install Mosquitto
+		2. Install Libmosquitto1
+		3. Install Collectd-core
+		4. thin-edge.io installation
+		5. Install Tedge mapper
+		6. Install Tedge agent
+		7. Install Tedge apama plugin
+		8. Install tedge apt plugin
+		9. Install Tedge logfile request plugin
+		10. Install C8y plugin
```
Comment (rina23q) [inline tests/RobotFramework/README.md:66]: It's a minor correction.
```suggestion
		9. Install C8y log plugin
		10. Install C8y configuration plugin
```

File: tests/RobotFramework/README.md [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+		11. Install Watchdog
```
Comment (rina23q) [inline tests/RobotFramework/README.md:67]:
```suggestion
		11. Install Tedge watchdog
```

File: tests/RobotFramework/README.md [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+		13. Set c8y URL
+		14. Upload certificate 
+		15. Connect to c8y
+
+		4.2. if yes than thin-edge.io will be uninstalled using the uninstallation script to purge it and than steps from 4.1 will take place
+	
+	Example command to run the task: 
+	robot -d \results --timestampoutputs --log health_tedge_mapper.html --report NONE --variable HOST:192.168.1.110 health_tedge_mapper.robot
+
+### 4.2. Test Cases
```
Comment (rina23q) [inline tests/RobotFramework/README.md:78]: It's nice to have such list in README. On the other hand, you need to keep it in mind that you always need to update README when you add a new test file.

**Review by rina23q — COMMENTED** [review]
Finished whole review.

File: tests/RobotFramework/E2E_Tutorials/Connect_to_c8y.robot [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+
+*** Variables ***
+${HOST}           
+${USERNAME}       pi
+${PASSWORD}       crypt:LO3wCxZPltyviM8gEyBkRylToqtWm+hvq9mMVEPxtn0BXB65v/5wxUu7EqicpOgGhgNZVgFjY0o=          
+${Version}        0.*
+${download_dir}    /home/pi/
+${url_dow}    https://github.com/thin-edge/thin-edge.io/actions
+# ${user_git}    crypt:3Uk76kNdyyYOXxus2GFoLf8eRlt/W77eEkcSiswwh04HNfwt0NlJwI7ATKPABmxKk8K1a8NsI5QH0w8EmT8GWeqrFwX2    
+# ${pass_git}    crypt:IcTs6FyNl16ThjeG6lql0zNTsjCAwg5s6PhjRrcEwQ9DVHHRB4TjrGcpblR6R1v7j9oUlL3RzwxGpfBfsijVnQ==
```
[the remark this answers is not in the document]
Comment (gligorisaev) [reply tests/RobotFramework/E2E_Tutorials/Connect_to_c8y.robot:23]: no, they are crypting my username and password for git, I don't want to be visible which account is downloading the build

File: tests/RobotFramework/E2E_Tutorials/Connect_to_c8y.robot [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+Library    CryptoLibrary    variable_decryption=True
+Library    Dialogs
+Library    String
+Library    CSVLibrary
+Library    OperatingSystem
+Suite Setup            Open Connection And Log In
+Suite Teardown         SSHLibrary.Close All Connections
+
+*** Variables ***
+${HOST}
```
[the remark this answers is not in the document]
Comment (gligorisaev) [reply tests/RobotFramework/E2E_Tutorials/Connect_to_c8y.robot:16]: Capital letters in variables means that this global variable is filled in by the command line but can also have an default value if needed

File: tests/RobotFramework/MQTT_health_check/health_c8y-configuration-plugin.robot [hunk 9]
```diff
@@ -0,0 +1,54 @@
+#Command to execute:    robot -d \results --timestampoutputs --log health_c8y-configuration-plugin.html --report NONE --variable HOST:192.168.1.120 /Users/glis/thin-edge.io-fork/tests/RobotFramework/MQTT_health_check/health_c8y-configuration-plugin.robot
```
[the remark this answers is not in the document]
Comment (gligorisaev) [reply tests/RobotFramework/MQTT_health_check/health_c8y-configuration-plugin.robot:1]: path needs to be corrected as soon I will have the first bunch, generally all commented lines are examples in most of the cases, they are in no way part of the testrun

[the remark this answers is not in the document]
Comment (rina23q) [reply tests/RobotFramework/MQTT_health_check/health_c8y-configuration-plugin.robot:1]: Ok, here we agreed to keep the path to forked repository. Later (after merge) we need to fix.

File: tests/RobotFramework/E2E_Tutorials/Connect_to_c8y.robot (hunk 7 above)
[the remark this answers is not in the document]
Comment (rina23q) [reply tests/RobotFramework/E2E_Tutorials/Connect_to_c8y.robot:23]: But they are commented out. So you don't use them.

File: tests/RobotFramework/child_alarms/c8y_child_alarms_rpi.robot [hunk 10]
```diff
@@ -0,0 +1,199 @@
...
+Library    CryptoLibrary    variable_decryption=True
+Suite Setup            Open Connection And Log In
+Suite Teardown         SSHLibrary.Close All Connections
+
+*** Variables ***
+${HOST}    
+${USERNAME}    pi
+${PASSWORD}    crypt:LO3wCxZPltyviM8gEyBkRylToqtWm+hvq9mMVEPxtn0BXB65v/5wxUu7EqicpOgGhgNZVgFjY0o=    #thinedge
+${url}    https://thin-edge-io.eu-latest.cumulocity.com/
+${url_tedge}    thin-edge-io.eu-latest.cumulocity.com
```
[the remark this answers is not in the document]
Comment (rina23q) [reply tests/RobotFramework/child_alarms/c8y_child_alarms_rpi.robot:18]: As of now, @gligorisaev and I decided that we don't care the url as we don't have the correct environment.

**Review by rina23q — APPROVED** [review]

**Pull request merged**